src/borg/repository.py

```diff
@@ -107,10 +107,14 @@ def __init__(self, id, repo):
     class InsufficientFreeSpaceError(Error):
         """Insufficient free space to complete transaction (required: {}, available: {})."""
 
-    def __init__(self, path, create=False, exclusive=False, lock_wait=None, lock=True, append_only=False):
+    class StorageQuotaExceeded(Error):
+        """The storage quota ({}) has been exceeded ({}). Try deleting some archives."""
```

Comment:
The generic error handling in protocol v2 means that we don't have to do that unless the type of the exception is relevant, i.e. some code expects that exception. For user-facing error messages there is no need.